lib/parser.js

```diff
@@ -175,6 +176,22 @@ async function handleCircularRefs(refParser, parsedJSON, initialFormat, asyncapi
   parsedJSON[String(xParserCircle)] = true;
 }
 
+/**
+ * Creates (or reuses) a function that validates an AsyncAPI document based on the passed AsyncAPI version.
+ * 
+ * @private
+ * @param {Object} version AsyncAPI version.
+ * @returns {Function} Function that validates an AsyncAPI document based on the passed AsyncAPI version.
+ */
+function getValidator(version) {
+  let validate = ajv.getSchema(version);
+  if (!validate) {
+    ajv.addSchema(asyncapi[String(version)], version);
+    validate = ajv.getSchema(version);
+  }
+  return validate;
+}
+
```

Comment:
Is there a reason you dont use the same function, instead of duplicating it?

Reply:
> Would really like to get a test for this 🧐 Not sure how to achieve it though 🤔 Any thoughts?

I only see the way to do the integration tests and test the memory consumption for e.g. 1000 iteration of parsing.

> Is there a reason you dont use the same function, instead of duplicating it?

Eeee 😅  Do you have in mind separate function for spec and separate for schema? I thought about that, but I would like to have separate AJV instance for that, because validation of schemas is done as custom operation.